pipelines/appco/scripts/longhorn-setup.sh

```diff
@@ -146,85 +161,123 @@ generate_longhorn_yaml_manifest() {
   LONGHORN_CSI_LIVENESSPROBE_IMAGE=`grep -io "longhornio\/livenessprobe:.*$" "${MANIFEST_BASEDIR}/longhorn.yaml"| head -1 | sed -e 's/^"//' -e 's/"$//'`
 
   # replace longhorn images with custom images
-  sed -i 's#'${LONGHORN_MANAGER_IMAGE}'#'${CUSTOM_LONGHORN_MANAGER_IMAGE}'#' "${MANIFEST_BASEDIR}/longhorn.yaml"
-  sed -i 's#'${LONGHORN_ENGINE_IMAGE}'#'${CUSTOM_LONGHORN_ENGINE_IMAGE}'#' "${MANIFEST_BASEDIR}/longhorn.yaml"
+  if [[ -z "${REGISTRY_URL}" ]]; then
+    sed -i 's#'${LONGHORN_MANAGER_IMAGE}'#'${CUSTOM_LONGHORN_MANAGER_IMAGE}'#' "${MANIFEST_BASEDIR}/longhorn.yaml"
+    sed -i 's#'${LONGHORN_ENGINE_IMAGE}'#'${CUSTOM_LONGHORN_ENGINE_IMAGE}'#' "${MANIFEST_BASEDIR}/longhorn.yaml"
+  else
+    sed -i 's#'${LONGHORN_MANAGER_IMAGE}'#'${REGISTRY_URL}/${CUSTOM_LONGHORN_MANAGER_IMAGE}'#' "${MANIFEST_BASEDIR}/longhorn.yaml"
```

Comment:
Can these `sed`s be replaced with the `if [[ ! -z "${REGISTRY_URL}" ]];` section in `customize_longhorn_manifest_registry` function in `longhorn_manifest.sh`?

If we have to handle the different image repo names between `registry.suse.de/home/mbjoerkelund/orchid/containers/containers/` and `longhornio`, can we control it using a `longhorn_repo` variable like how we do in [`longhorn-rancher-chart-test` ](https://ci.longhorn.io/job/private/job/longhorn-rancher-chart-test/) to decide whether to pull images from `rancher` repo or `longhornio` repo?

Reply:
If it's feasible but we don't have the bandwidth to refine it, a new ticket can be created to track it.

Reply:
Updated and belwo is current code test result:
- Harvester amd64 : [test_basic.py::test_volume_basic](https://10.115.5.5/job/private/job/longhorn-appco-images-test/65/)
- AWS arm 64 : [test_basic.py::test_volume_basic](https://10.115.5.5/job/private/job/longhorn-appco-images-test/66/)

For the refinement part, it is applicable, I have created https://github.com/longhorn/longhorn/issues/10807 to track it, thank you.